### ISSUE TITLE
Distortion coefficient

### ISSUE THREAD

**Issue by ADAS-Iv** [kaustubh-sadekar/VirtualCam#3]
Hello,

I'm trying to use GUI.py but I'm having some troubles with distortion coefficient: I see variation in the chessboard pattern only when I put the coefficients at their max value

**Comment by lxzrv** [comment]
Hi! Faced with that problem too.
To fix it you should assign dtype of KpCoeff in vcam lib to any np.float type

> Hello,
> 
> I'm trying to use GUI.py but I'm having some troubles with distortion coefficient: I see variation in the chessboard pattern only when I put the coefficients at their max value

**Comment by vgrllc** [comment]
I just changed:

self.KpCoeff = np.array([0,0,0,0,0,0,0,0] 

to:

self.KpCoeff = np.array([0,0,0,0,0,0,0,0], dtype=float)

in the vcam.py source and it is working.

**Comment by b-hakim** [comment]
> I just changed:
> 
> self.KpCoeff = np.array([0,0,0,0,0,0,0,0]
> 
> to:
> 
> self.KpCoeff = np.array([0,0,0,0,0,0,0,0], dtype=float)
> 
> in the vcam.py source and it is working.

Works for me! 

Author need to make that change @kaustubh-sadekar

**Comment by truher** [comment]
+1 where's the vcam source?

**Comment by ahtsan** [comment]
You can do the following in terminal
```
python3
>>> import vcam
>>> import os
>>> os.path.dirname(vcam.__file__)
```

**Comment by ahtsan** [comment]
> +1 where's the vcam source?

You can do the following in terminal
```
python3
>>> import vcam
>>> import os
>>> os.path.dirname(vcam.__file__)
```

**Comment by truher** [comment]
indeed.  i meant "where is the vcam source in github so i can fix it for everyone?"

**Comment by henrikmidtiby** [comment]
I have managed to locate the source for the vcam python module and then added it to this fork of the project.

https://github.com/henrikmidtiby/VirtualCam/commit/918443dee4347594df31ad9a79066ff1dd9df788

**Comment by ArnobG2002** [comment]
> I just changed:
> 
> self.KpCoeff = np.array([0,0,0,0,0,0,0,0]
> 
> to:
> 
> self.KpCoeff = np.array([0,0,0,0,0,0,0,0], dtype=float)
> 
> in the vcam.py source and it is working.

You sure it's working accurately just like that? I doubt because the distortion is dependent on the Transformation Matrix calculated in the function `update_M()` and I think there is a huge mistake in the following block of code:
```python
Rx = np.array([[1, 0, 0], [0, math.cos(self.alpha), -math.sin(self.alpha)], [0, math.sin(self.alpha), math.cos(self.alpha)]])
Ry = np.array([[math.cos(self.beta), 0, -math.sin(self.beta)], [0, 1, 0], [math.sin(self.beta), 0, math.cos(self.beta)]])
Rz = np.array([[math.cos(self.gamma), -math.sin(self.gamma), 0], [math.sin(self.gamma), math.cos(self.gamma), 0], [0, 0, 1]])
self.R = np.matmul(Rx, np.matmul(Ry, Rz))
```
On what basis did you say that it is working? Is it simply because you see some distortion as you change the distortion coefficients through the trackbar? Have you verified if it is the correct distortion given your input distortion coefficients?
